pkg/storageincentives/staking/contract.go

```diff
@@ -278,7 +280,27 @@ func (c *contract) getBalance(ctx context.Context) (*big.Int, error) {
 	return abi.ConvertType(results[0], new(big.Int)).(*big.Int), nil
 }
 
-func (c *contract) WithdrawAllStake(ctx context.Context) (txHash common.Hash, err error) {
+func (c *contract) WithdrawStake(ctx context.Context) (txHash common.Hash, err error) {
+	stakedAmount, err := c.getStake(ctx)
+	if err != nil {
+		return
+	}
+
+	if stakedAmount.Cmp(big.NewInt(0)) <= 0 {
+		return common.Hash{}, ErrInsufficientStake
+	}
```

Comment:
use newly introduced `withdrawableStake()` instead.
this part is wrong because the contract does not allow to withdraw all stake, only the amount of which potential stake greater than effective stake